File: app/src/main/java/io/mgba/Adapters/TabViewPager.java
package io.mgba.Adapters;

import android.os.Bundle;
import android.support.v4.app.Fragment;
import android.support.v4.app.FragmentManager;
import android.support.v4.app.FragmentStatePagerAdapter;

import io.mgba.Constants;
import io.mgba.UI.Fragments.Main.FavouritesFragment;
import io.mgba.UI.Fragments.Main.GameFragment;

import static io.mgba.mgba.printLog;

public class TabViewPager extends FragmentStatePagerAdapter {
    private final static String TAG = "mgba:TabPager";
    private final int[] platforms = {Constants.PLATFORM_FAVS, Constants.PLATFORM_GBA, Constants.PLATFORM_GBC};

    public TabViewPager(FragmentManager fm) {
        super(fm);
        printLog(TAG, "CTOR");
    }

    @Override
    public Fragment getItem(int position) {
        Bundle args = new Bundle();
        args.putInt(Constants.ARG_PLATFORM, platforms[position]);

        final GameFragment fragment = position == 0 ? new FavouritesFragment() : new GameFragment();
        fragment.setArguments(args);
        return fragment;

    }

    @Override
    public int getCount() {
        return platforms.length;
    }
}

File: app/src/main/java/io/mgba/main/adapters/TabViewPager.kt
package io.mgba.main.adapters

import android.os.Bundle
import androidx.fragment.app.Fragment
import androidx.fragment.app.FragmentManager
import androidx.fragment.app.FragmentStatePagerAdapter
import io.mgba.main.fragments.FavouritesFragment
import io.mgba.main.fragments.GameFragment
import io.mgba.mgba.Companion.printLog

class TabViewPager(fm: FragmentManager) : FragmentStatePagerAdapter(fm) {


    private val platforms = intArrayOf(PLATFORM_FAVS, PLATFORM_GBA, PLATFORM_GBC)

    init {
        printLog(TAG, "CTOR")
    }

    override fun getItem(position: Int): Fragment {
        val args = Bundle()
        args.putInt(ARG_PLATFORM, platforms[position])

        val fragment = if (position == 0) FavouritesFragment() else GameFragment()
        fragment.arguments = args
        return fragment

    }

    override fun getCount(): Int {
        return platforms.size
    }

    companion object {
        private val TAG = "mgba:TabPager"
        val PLATFORM_GBC = 0
        val ARG_PLATFORM = "platform"
        val PLATFORM_GBA = 1
        val PLATFORM_FAVS = 2
    }
}
